WcaOnRails/spec/models/competition_spec.rb

```diff
@@ -311,33 +311,41 @@
     end
   end
 
-  describe "when confirming" do
-    let(:competition) { FactoryGirl.create :competition, :with_delegate }
+  describe "when confirming or making visible" do
+    let(:competition_with_delegate) { FactoryGirl.build :competition, :with_delegate }
+    let(:competition_without_delegate) { FactoryGirl.build :competition }
 
-    it "works" do
-      competition.isConfirmed = true
-      expect(competition).to be_valid
-    end
+    [:isConfirmed, :showAtAll].each do |action|
+      it "can set #{action}" do
+        competition_with_delegate.public_send "#{action}=", true
+        expect(competition_with_delegate).to be_valid
+      end
 
-    [:cityName, :countryId, :venue, :venueAddress, :website, :latitude, :longitude].each do |field|
-      it "requires #{field}" do
-        competition.public_send "#{field}=", ""
-        competition.isConfirmed = true
-        expect(competition).not_to be_valid
+      [:cityName, :countryId, :venue, :venueAddress, :website, :latitude, :longitude].each do |field|
+        it "requires #{field} when setting #{action}" do
+          competition_with_delegate.public_send "#{field}=", ""
+          competition_with_delegate.public_send "#{action}=", true
```

Comment:
Such an assignment is a bit cumbersome. Moreover there's a handy way to write these two lines at once:
```ruby
competition_with_delegate.assign_attributes field => "", action => true
```